queue_job/job.py

```diff
@@ -517,6 +542,9 @@ def perform(self):
 
         The job is executed with the user which has initiated it.
         """
+        if self.max_retries and self.retry >= self.max_retries:
+            raise FailedJobError("Max. retries (%d) reached" % (self.max_retries))
```

Comment:
Why adding this? This sounds unrelated.

Reply:
The check that `max_retries` is greater than `retry` was done only  when a `RetryableJobError` was raised during the execution of the job.

Since I am increasing the `retry` when re-queuing a job that was killed, I need to do the check before peforming the job, not only when raising an exception.

Moreover, I think that it doesn't make sense to perform the check only when an exception has been raised :thinking: